client-react/src/pages/static-app/configuration/ConfigurationSnippets.tsx

```diff
@@ -297,31 +300,35 @@ const ConfigurationSnippets: React.FC<ConfigurationSnippetsProps> = ({
     );
   };
 
-  const isDeleteButtonEnabled: boolean = useMemo(() => {
-    return disabled || !values.snippets || values.snippets?.filter(snippet => snippet.checked).length <= 0;
-  }, [values.snippets]);
+  const isDeleteButtonDisabled: boolean = useMemo(() => {
+    return !hasWritePermissions || !values.snippets || values.snippets?.filter(snippet => snippet.checked).length <= 0;

```

Comment:
"?" is redundant since you are already checking "values.snippets" before 
```suggestion
    return !hasWritePermissions || !values.snippets || values.snippets.filter(snippet => snippet.checked).length == 0;

```